src/security/Session.h

```diff
@@ -18,6 +18,31 @@
 #if USE_OPENSSL
 #if HAVE_OPENSSL_SSL_H
 #include <openssl/ssl.h>
+
+/// These are not included by openssl/ssl.h when deprecated API is disabled
+#include <openssl/opensslv.h>
+#include <openssl/bn.h>
+#include <openssl/rsa.h>
+#include <openssl/dh.h>
+#include <openssl/x509.h>
+
+/// Use new calls for API deprecated in OpenSSL 1.1.0
+/// Note that all of the calls in this next group were implemented at the same time
+/// as X509_STORE_CTX_set0_untrusted, in both OpenSSL 1.1.0, and LibreSSL 2.7.0.
+#if !HAVE_LIBCRYPTO_X509_STORE_CTX_SET0_UNTRUSTED
+#define OPENSSL_VERSION SSLEAY_VERSION
```

Comment:
NP: OpenSSL version and SSLEAY version are not necessarily the same thing.

If (as I suspect) the removal of SSLEAY version lookup is related to the relicensing of OpenSSL. Then the restricted builds may have different license requirements - maybe avoiding the need for the code using that call. Please check that if you can, or this may get held up waiting for someone else (us?) to check.

Please separate these *_version() define from the X509 definitions regardless. A separate check in lib-checks.m4 should solve the issue of LibreSSL differences mentioned above and also cope with any compatibility definitions OpenSSL may have been built with.

Reply:
I'll have to check the licensing terms, but it should not be dependend on what names are being used.
`OPENSSL_VERSION` is the new name of the flag passed to `OpenSSL_version()` to return the text variant of the version number and the release date. They were renamed from `SSLEAY_VERSION` and `SSLeay_version()`, respectively. See https://www.openssl.org/docs/man1.1.0/crypto/OpenSSL_version.html, and compare it to https://www.openssl.org/docs/man1.0.2/crypto/SSLeay_version.html.
I can add a check for `OpenSSL_version()` in `lib-checks.m4`, but, like I said, they were all implemented at the same time, so it will cause an unnecessary check when configuring.

Reply:
There was only one line using both of these symbols, so I can just check if `SSLEAY_VERSION` is defined. I've checked in libressl (2.6,2.7,2.8) and openssl (1.02, 1.1.0 and 1.1.1), and it is always defined as 0, so it should be safe to use it as a check for which version to use.